src/ert/_c_wrappers/enkf/model_config.py

```diff
@@ -40,11 +40,29 @@ def __init__(
             else self.DEFAULT_HISTORY_SOURCE
         )
 
-        self.runpath_format_string = (
-            runpath_format_string
-            if runpath_format_string is not None
-            else self.DEFAULT_RUNPATH
-        )
+        self.has_valid_runpath = False
+
+        if runpath_format_string is None:
+            self.runpath_format_string = self.DEFAULT_RUNPATH
+            self.has_valid_runpath = True
+        elif "%d" in runpath_format_string:
+            self.runpath_format_string = runpath_format_string
+            self.has_valid_runpath = True
+            logger.warning(
+                "RUNPATH keyword should use syntax "
+                f"`{self.DEFAULT_RUNPATH}` "
+                "instead of deprecated syntax "
+                f"`{runpath_format_string}`"
+            )
+        else:
+            self.runpath_format_string = runpath_format_string
+            logger.error(
+                "RUNPATH keyword should use syntax "
```

Comment:
The error message looks a bit contradicting. Do we want %d or not?.

Reply:
Indeed. New suggestion!